### ISSUE TITLE
Missing else, improved calculation

### ISSUE THREAD

**Issue by hansiglaser** [Zaaphod/Ethernet-HTTP-Relays#1]
I just got curious after I saw you post in the FreePascal mailing list and looked at your code.

https://github.com/Zaaphod/Ethernet-HTTP-Relays/blob/79ce11ab60ab337ee700b29fd000a15f7ce00e0f/ethernet_relays.pas#L556

What if Relay_Number is <1 or >16? Then Needed_Page stays uninitialized. Fortunately it is checked in line 539 (given that Count is <= 16).

In addition, the if-else-tree can be simplified to
`  Needed_Page := 1 + ((Relay_Number-1) shr 2);`
or even simpler
`  Needed_Page := (Relay_Number+3) shr 2;`
and then doesn't need any additional checks.

**Comment by Zaaphod** [comment]
Thank you for looking so close at this project, I appreciate your suggestion and will implement it when I get back from traveling.

**Comment by Zaaphod** [comment]
@hansiglaser I've implemented your suggestion
https://github.com/Zaaphod/Ethernet-HTTP-Relays/blob/4982388cf81b7c853af7cd2e7c1c8a43f42222bc/ethernet_relays.pas#L447

I also found that I could do something similar on line 171 where I had a similar structure.
https://github.com/Zaaphod/Ethernet-HTTP-Relays/blob/4982388cf81b7c853af7cd2e7c1c8a43f42222bc/ethernet_relays.pas#L171

I also cleaned things up a bit and made functions for duplicate operations.

I only tested it quickly to prove that everything still functions, I didn't extensively test what happens if I put in invalid relay numbers or things like that yet.

Thanks again for the elegant suggestion!
